src/utils/roomVersion.ts

```diff
@@ -0,0 +1,35 @@
+/*
+Copyright 2025 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/**
+ * Checks if the given room version is a version that is known to use
+ * new "hydra" power level semantics (ie. room version 12 or later)
+ * (see https://github.com/matrix-org/matrix-spec-proposals/pull/4289).
+ * Note that this will only return true for versions that are known to
+ * the js-sdk': any room version created subsequently will need to be added here
+ * if it uses hydra semantics.
+ *
+ * Once hydra has been rolled out in production for "long enough", it would probably
+ * makes sense to assume hydra semantics by default and hence check instead for room
+ * versions known to use the old semantics using a different function (which does not exist
+ * currently).
```

Comment:
Sorry, but this still feels like the target audience is a future js-sdk maintainer, not an application developer.

"will need to be added here"

"it would probably makes [sic] sense to"

```suggestion
 *
 * Note that this will only return `true` for versions that are known to
 * the js-sdk.
 *
 * In future, once hydra has been in production for some time, this function will likely
 * be updated to reverse the logic, and return `true` for all room versions except those
 * known to use the old semantics.
```

... or "this function will be removed in favour of another that uses the opposite semantics", or something.

Reply:
Possibly, yeah - this would mean we break any room versions we don't know about (eg. experimental ones) as we'll start treating them as hydra, although this will happen at some point anyway so arguably it may as well happen now. In the interest of trying to move faster, I've changed it to do this to see what folks think.